hibernate-core/src/main/java/org/hibernate/engine/transaction/jta/platform/internal/AbstractJtaPlatform.java

```diff
@@ -31,12 +31,20 @@ public abstract class AbstractJtaPlatform
 	private boolean cacheUserTransaction;
 	private ServiceRegistryImplementor serviceRegistry;
 
-	private final JtaSynchronizationStrategy tmSynchronizationStrategy = new TransactionManagerBasedSynchronizationStrategy( this );
+	private final TransactionManagerAccessBridge transactionManagerAccess = new TransactionManagerAccessBridge();
+
+	private final JtaSynchronizationStrategy tmSynchronizationStrategy = new TransactionManagerBasedSynchronizationStrategy( transactionManagerAccess );
```

Comment:
I'd suggest simply inlining the implementation of `TransactionManagerBasedSynchronizationStrategy` into `AbstractJtaPlatform`. It's all internal, so no worries about backwards compatibility.

Reply:
I inlined it as an inner class, and I really prefer that over having the platform implementing the interface.